src/ParseFasm.cpp

```diff
@@ -0,0 +1,546 @@
+// Copyright (C) 2020  The SymbiFlow Authors.
+//
+// Use of this source code is governed by a ISC-style
+// license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/ISC
+//
+// SPDX-License-Identifier: ISC
+
+#include "antlr4-runtime.h"
+#include "FasmLexer.h"
+#include "FasmParser.h"
+#include "FasmParserVisitor.h"
+#include "Tests.h"
+
+/// This code parses FASM and produces a lightweight binary format that
+/// is fast and simple to unpack based on the tag/length/value (TLV)
+/// format.
+///
+/// For example, for a fixed width integer, this can just be:
+///   <tag : 1 byte> <data : 4 bytes>
+/// and for a variable length string:
+///   <tag : 1 byte> <length : 4 bytes> <data : length bytes>
+///
+/// Note that the length itself must be a fixed width value, which does
+/// impose a size limit, but this format is more efficient and easier
+/// to decode than a UTF-8 style variable length encoding where a bit
+/// is reserved per per byte to indicate the end of a variable length
+/// value.
+///
+/// TLVs can be nested, with each level adding 55 bytes of header
```

Comment:
That should be `5 bytes`, not `55 bytes`, yes?

Reply:
Yes, thanks. Fixed.